osu.Game/Online/Leaderboards/LeaderboardScore.cs

```diff
@@ -356,6 +346,24 @@ public ScoreComponentLabel(LeaderboardScoreStatistic statistic)
             }
         }
 
+        private class RankLabel : Container, IHasTooltip
+        {
+            public RankLabel(int? rank)
+            {
+                TooltipText = rank == null || rank < 1000 ? null : $"#{rank:N0}";
```

Comment:
```suggestion
                if (rank > 1000)
                    TooltipText = $"#{rank:N0}";
```

Reply:
I've edited your suggestion as we also need 1000 to be included. `ToMetric` will format numbers 1000-1050 to `1k`.